src/Routing/Controllers/ZaakInformationObjectRoutingController.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OWC\Zaaksysteem\Routing\Controllers;
+
+use Exception;
+
+class ZaakInformationObjectRoutingController extends AbstractRoutingController
+{
+    public function register(): void
+    {
+        $this->addCustomRewriteRules();
+        $this->allowCustomQueryVars();
+        $this->includeObjectInTemplate();
+    }
+
+    /**
+     * Implement custom routing for routing for downloading 'zaak' information objects.
+     *
+     * The implementation for downloading 'zaak' information objects requires:
+     * - a page with 'zaak-download' as the slug
+     * - the page to be requested with an identification and supplier in the URI
+     */
+    protected function addCustomRewriteRules(): void
+    {
+        add_action('init', function () {
+            flush_rewrite_rules();
+
+            add_rewrite_rule(
+                'zaak-download/([a-zA-Z0-9-]+)/([a-zA-Z]+)/?$',
+                'index.php?pagename=zaak-download&zaak_download_identification=$matches[1]&zaak_supplier=$matches[2]',
+                'top'
+            );
+        });
+    }
+
+    /**
+     * Add requested 'zaak' and information object vars to the query variables.
+     */
+    protected function allowCustomQueryVars(): void
+    {
+        add_action('query_vars', function (array $queryVars) {
+            $queryVars[] = 'zaak_supplier';
+            $queryVars[] = 'zaak_download_identification';
+
+            return array_unique($queryVars);
+        });
+    }
+
+    protected function includeObjectInTemplate(): void
+    {
+        add_action('template_include', function ($template) {
+            $this->handleZaakDownload();
+
+            return $template;
+        }, 999, 1); // High priority so the validateTemplate method inside the ValidationServiceProvider runs first.
+    }
+
+    protected function handleZaakDownload(): void
```

Comment:
Moet hier ook nog iets van een authenticatie check plaatsvinden? Of wordt dat ergens anders al afgehandeld? M.a.w. mag de huidige user wel dit bestand downloaden?

Reply:
Yes in de methode erboven staat: 
```
        add_action('template_include', function ($template) {
            $this->handleZaakDownload();

            return $template;
        }, 999, 1); // High priority so the validateTemplate method inside the ValidationServiceProvider runs first.
```

Reply:
Top! Maar is dan ook zeker dat de huidige user _dat_ documen in mag zien? 

Ik zou in theorie kunnen itereren over de document identifiers en zo alle documenten in het zaaksysteem in kunnen zien. Volgens mij zou je de identifier voor de zaak (waar het document bij hoort) ook in de URL moeten zetten, zodat je kan checken dat de gebruiker toegang heeft tot die zaak én dat het opgevraagde document ook bij die zaak hoort.

Als laatste mis ik nog een check op de vertrouwelijkheidaanduiding. Ik ben even kwijt welke aanduidingen aangeven dat een gebruiker het document mag in zien. Maar @EduardWitteveen weet dit wel :)

Reply:
Goed bezig, ook fijn om te lezen dat er wordt nagedacht over een Bridge pattern, lijkt me de juiste aanpak om de code uitwisselbaar te houden met verschillende zgw-servers/zgw-versies. 

Qua bevraging zou ik verwachten dat de check altijd een combinatie van zaak/document waarbij zaak-initiator-rol altijd wordt gecontroleerd. Vanuit de zaak zou je denken dat je via de url wel het document kan opvragen. In mijn ogen is deze url een interne url en is de (document)identificatie de te delen publieke identificatie.
Dan kom je er waarschijnlijk op uit dat de mapping iets als /zaken/1234/documenten/5678 uitkomt, waarbij het identificaties zijn.

Owja, welke controles zou je dan willen hebben?

Voor de zaak:
- zaaktype-identificatie: Alleen deze zaaktypen mogen ontsloten worden naar inwoner, voorbeeld: B0208
- zaaktype-ingangsdatum: Alleen zaken met startdatum na deze datum mogen ontsloten worden naar de inwoner, voorbeeld: "2023-11-09"
- zaak-vertrouwelijkheidaanduiding: Deze mogen in mijn ogen alleen zaken van: "openbaar" / "beperkt_openbaar" /  "zaakvertrouwelijk" / "vertrouwelijk" zijn, niet "intern" / "confidentieel" / "geheim" "zeer_geheim"
- zaak.rol[omschrijvingGeneriek='initiator' && betrokkeneType='natuurlijk_persoon'].betrokkeneIdentificatie.inpBsn = %bsnnummer%

Voor de document:
- document.vertrouwelijkheidaanduiding : Deze mogen in mijn ogen alleen document van: "openbaar" / "beperkt_openbaar" /  "zaakvertrouwelijk" / "vertrouwelijk" zijn, niet "intern"/ "confidentieel" / "geheim" / "zeer_geheim"
- document.status : Deze mogen zijn: "definitief" / "gearchiveerd" en niet: "in_bewerking" / "ter_vaststelling" 
- document.zaak : Altijd controleren of de zaak ook bekeken mag worden, rechten worden bepaald vanuit de zaak-contetx.

Reply:
Een deel heb ik toegevoegd maar laat ik nog even openstaan voor het volgende fieldlab. Ik denk dat als we alle zaken gewoon ophalen met de filter op BSN aan, nu is het een optie om deze filtering uit te zetten, we een hoop afvangen. Los van de vertrouwelijkheidaanduidingen en statussen. Laten we op 11-12 even zitten om dit goed door te spreken.